TILApp/ViewModels/BlogViewModel.swift

```diff
@@ -55,13 +55,16 @@ final class BlogViewModel {
             onError?(error)
         }
     }
-    
-    // TODO: 결과 받아와서 blogs에 적용하기
+
     func setMainBlog(_ id: Int, onSuccess: (() -> Void)? = nil, onError: ((Error) -> Void)? = nil) {
         APIService.shared.request(.setMainBlog(id)) { [weak self] _ in
-            guard let self else { return }
-            
-            onSuccess?()
+            guard let self = self else { return }
+            load { [weak self] _ in
+                guard let self else { return }
+                onSuccess?()
+            } onError: { error in
+                print(error)
+            }
```

Comment:
```suggestion
         APIService.shared.request(.setMainBlog(id), model: Blog.self) { [weak self] blog in
            guard let self else { return }
            // blogs에서 기존 main 블로그의 main을 false로 전환,
            // 변경한 blog의 main만 true로 변경하시면 될 듯합니다.

```

굳이 새로 불러올 필요 없을 거 같아요.